apps/prairielearn/src/pages/instructorAssessmentSettings/instructorAssessmentSettings.ts

```diff
@@ -119,6 +148,86 @@ router.post(
           tid_new,
         });
 
+        const serverJob = await editor.prepareServerJob();
+        try {
+          await editor.executeWithServerJob(serverJob);
+          return res.redirect(req.originalUrl);
+        } catch {
+          return res.redirect(res.locals.urlPrefix + '/edit_error/' + serverJob.jobSequenceId);
+        }
+      }
+    } else if (req.body.__action === 'update_assessment') {
+      const infoAssessmentPath = encodePath(
+        path.join(
+          res.locals.course.path,
+          'courseInstances',
+          res.locals.course_instance.short_name,
+          'assessments',
+          res.locals.assessment.tid,
+          'infoAssessment.json',
+        ),
+      );
+      if (!(await fs.pathExists(infoAssessmentPath))) {
+        throw new error.HttpStatusError(400, 'infoAssessment.json does not exist');
+      }
+      const paths = getPaths(undefined, res.locals);
+
+      const assessmentInfo = JSON.parse(await fs.readFile(infoAssessmentPath, 'utf8'));
+      const assessmentInfoEdit = assessmentInfo;
+      assessmentInfoEdit.title = req.body.title;
+      assessmentInfoEdit.set = req.body.set;
+      assessmentInfoEdit.number = req.body.number;
+      assessmentInfoEdit.module = req.body.module;
+      const formattedJson = await formatJsonWithPrettier(JSON.stringify(assessmentInfoEdit));
+
+      const editor = new FileModifyEditor({
+        locals: res.locals,
+        container: {
+          rootPath: paths.rootPath,
+          invalidRootPaths: paths.invalidRootPaths,
+        },
+        filePath: infoAssessmentPath,
+        editContents: b64EncodeUnicode(formattedJson),
+        origHash: req.body.orig_hash,
+      });
+      const serverJob = await editor.prepareServerJob();
+      try {
+        await editor.executeWithServerJob(serverJob);
+        flash('success', 'Assessment updated successfully');
+      } catch {
+        return res.redirect(res.locals.urlPrefix + '/edit_error/' + serverJob.jobSequenceId);
+      }
+
+      // Assessment TID must be handled separately
+      if (!req.body.aid) {
+        throw new error.HttpStatusError(400, `Invalid TID (was falsy): ${req.body.aid}`);
+      }
+      if (!/^[-A-Za-z0-9_/]+$/.test(req.body.aid)) {
+        throw new error.HttpStatusError(
+          400,
+          `Invalid TID (was not only letters, numbers, dashes, slashes, and underscores, with no spaces): ${req.body.id}`,
+        );
+      }
+      if (res.locals.assessment.tid === req.body.aid) {
+        return res.redirect(req.originalUrl);
+      }
+      let tid_new;
+      try {
+        tid_new = path.normalize(req.body.aid);
+      } catch {
+        throw new error.HttpStatusError(
+          400,
+          `Invalid TID (could not be normalized): ${req.body.aid}`,
+        );
+      }
+      if (res.locals.assessment.tid === tid_new) {
```

Comment:
Consider inverting the conditional so you can write things like this:

```js
if (res.locals.assessment.tid !== tid_new) {
  doRenameHere();
}

// Maybe add a flash message here?
res.redirect(req.originalUrl);
```

Reply:
Inverted the conditional in [ca2ca5d](https://github.com/PrairieLearn/PrairieLearn/pull/10677/commits/ca2ca5d0e50c8f76d05e06333bbe2e8d3c49cf1c). 
Also, while working on this I realized how easy it can be to add spaces. I added a conditional to the client script to disable the save button if the AID includes a space.